### PR TITLE
KIBA-1758: Create tbl_sub_collection_tokens.

### PR DESCRIPTION
**Pull request by Femi-Ogunkola**

### CODE REVIEW

File: dbschema/3.users.sql [hunk 1]
```diff
@@ -48,6 +48,8 @@ GRANT INSERT, SELECT, UPDATE, DELETE ON tbl_gallery_badge_assignments TO notd_ap
 GRANT ALL ON SEQUENCE tbl_gallery_badge_assignments_id_seq TO notd_api;
 GRANT INSERT, SELECT, UPDATE, DELETE ON tbl_token_stakings TO notd_api;
 GRANT ALL ON SEQUENCE tbl_token_stakings_id_seq TO notd_api;
+GRANT INSERT, SELECT, UPDATE, DELETE ON tbl_sub_collection_tokens TO notd_api;
```
Comment (krishan711) [inline dbschema/3.users.sql:51]: i dont think this needs to delete

File: dbschema/1.tables.sql [hunk 2]
```diff
@@ -509,3 +509,44 @@ CREATE INDEX tbl_token_stakings_owner_address ON tbl_token_stakings (owner_addre
...
+
+CREATE TABLE tbl_sub_collection_tokens (
+    id BIGSERIAL PRIMARY KEY,
+    created_date TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+    updated_date TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+    registry_address TEXT NOT NULL,
+    sub_collection_id BIGINT NOT NULL,
+    token_id TEXT NOT NULL
+);
+
```
Comment (krishan711) [inline dbschema/1.tables.sql:521]: no space between tbl and indices

File: dbschema/1.tables.sql [hunk 3]
```diff
@@ -509,3 +509,44 @@ CREATE INDEX tbl_token_stakings_owner_address ON tbl_token_stakings (owner_addre
...
+    url TEXT,
+    discord_url TEXT,
+    banner_image_url TEXT,
+    does_support_erc721 BOOLEAN NOT NULL,
+    does_support_erc1155 BOOLEAN NOT NULL
+);
+CREATE UNIQUE INDEX tbl_sub_collections_registry_address_external_id ON tbl_sub_collections (registry_address, external_id);
+CREATE INDEX tbl_sub_collections_registry_address_updated_date ON tbl_sub_collections (registry_address, updated_date);
+CREATE INDEX tbl_sub_collections_updated_date ON tbl_sub_collections (updated_date);
+CREATE INDEX tbl_sub_collections_name ON tbl_sub_collections (name);
```
Comment (krishan711) [inline dbschema/1.tables.sql:552]: newline at end of file

File: dbschema/1.tables.sql [hunk 4]
```diff
@@ -509,3 +509,44 @@ CREATE INDEX tbl_token_stakings_owner_address ON tbl_token_stakings (owner_addre
...
+    wiki_url TEXT,
+    opensea_slug TEXT,
+    url TEXT,
+    discord_url TEXT,
+    banner_image_url TEXT,
+    does_support_erc721 BOOLEAN NOT NULL,
+    does_support_erc1155 BOOLEAN NOT NULL
+);
+CREATE UNIQUE INDEX tbl_sub_collections_registry_address_external_id ON tbl_sub_collections (registry_address, external_id);
+CREATE INDEX tbl_sub_collections_registry_address_updated_date ON tbl_sub_collections (registry_address, updated_date);
```
Comment (krishan711) [inline dbschema/1.tables.sql:550]: these should have pretty much all the same indices as collections

File: api/notd/sub_collection_token_processor.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    async def retrieve_sub_collection_name(self, registryAddress: str, tokenId: str) -> SubCollectionKey:
+        collectionName = None
+        if registryAddress == OPENSEA_SHARED_STOREFRONT_ADDRESS:
+            tokenAssetUrl = f'https://api.opensea.io/api/v1/asset/{registryAddress}/{tokenId}/'
+            tokenAssetResponse = await self.openseaRequester.get(url=tokenAssetUrl, timeout=10)
+            tokenAssetDict = tokenAssetResponse.json()
+            collectionName = tokenAssetDict.get('collection', {}).get('slug', None)
+            return SubCollectionKey(registryAddress=registryAddress, externalId=collectionName)
+        return
```
Comment (krishan711) [inline api/notd/sub_collection_token_processor.py:21]: raise exception

File: api/notd/model.py [hunk 6]
```diff
@@ -717,3 +723,45 @@ class UserTradingOverview:
...
+    openseaSlug: Optional[str]
+    url: Optional[str]
+    discordUrl: Optional[str]
+    bannerImageUrl: Optional[str]
+    doesSupportErc721: bool
+    doesSupportErc1155: bool
+
+
+@dataclasses.dataclass
+class SubCollection(RetrievedSubCollection):
```
Comment (krishan711) [inline api/notd/model.py:758]: put subcollection above subcollectiontoken in all contexts

File: dbschema/1.tables.sql (hunk 4 above)
Comment (Femi-Ogunkola) [reply dbschema/1.tables.sql:533]: It does, I copied and pasted

**Pull request merged**